test/constants/test_constants.f90

```diff
@@ -0,0 +1,408 @@
+    implicit none
+    type(error_type), allocatable, intent(out) :: error 
+    real(dp) :: value, expected, diff, fac
+    fac = 1.0d0
+    expected = 0.0d0 * fac
+    value = ELECTRON_VOLT%to_real(fac, uncertainty=.true.) * fac
+    diff = expected - value
+    call check(error, diff, 0.0d0)
+    if (allocated(error)) return
+end subroutine
+
+subroutine test_U_ELEMENTARY_CHARGE(error)
+    implicit none
+    type(error_type), allocatable, intent(out) :: error 
+    real(dp) :: value, expected, diff, fac
+    fac = 1.0d0
+    expected = 0.0d0 * fac
+    value = ELEMENTARY_CHARGE%to_real(fac, uncertainty=.true.) * fac
+    diff = expected - value
+    call check(error, diff, 0.0d0)
+    if (allocated(error)) return
+end subroutine
+
+subroutine test_U_FARADAY_CONSTANT(error)
+    implicit none
+    type(error_type), allocatable, intent(out) :: error 
+    real(dp) :: value, expected, diff, fac
+    fac = 1.0d0
+    expected = 0.0d0 * fac
+    value = FARADAY_CONSTANT%to_real(fac, uncertainty=.true.) * fac
+    diff = expected - value
+    call check(error, diff, 0.0d0)
+    if (allocated(error)) return
+end subroutine
+
+subroutine test_U_MOLAR_MASS_CONSTANT(error)
+    implicit none
+    type(error_type), allocatable, intent(out) :: error 
+    real(dp) :: value, expected, diff, fac
+    fac = 1.0d13
+    expected = 0.00000000031d-3 * fac
+    value = MOLAR_MASS_CONSTANT%to_real(fac, uncertainty=.true.) * fac
+    diff = expected - value
+    call check(error, diff, 0.0d0)
+    if (allocated(error)) return
+end subroutine
+
+subroutine test_U_MOLAR_VOLUME_NTP(error)
+    implicit none
+    type(error_type), allocatable, intent(out) :: error 
+    real(dp) :: value, expected, diff, fac
+    fac = 1.0d0
+    expected = 0.0d0 * fac
+    value = MOLAR_VOLUME_OF_IDEAL_GAS_273_15_K_101_325_KPA%to_real(fac, uncertainty=.true.) * fac
+    diff = expected - value
+    call check(error, diff, 0.0d0)
+    if (allocated(error)) return
+end subroutine
+
+subroutine test_U_PLANCK_CONSTANT(error)
+    implicit none
+    type(error_type), allocatable, intent(out) :: error 
+    real(dp) :: value, expected, diff, fac
+    fac = 1.0d0
+    expected = 0.0d0 * fac
+    value = PLANCK_CONSTANT%to_real(fac, uncertainty=.true.) * fac
+    diff = expected - value
+    call check(error, diff, 0.0d0)
+    if (allocated(error)) return
+end subroutine
+
+subroutine test_U_SPEED_OF_LIGHT(error)
+    implicit none
+    type(error_type), allocatable, intent(out) :: error 
+    real(dp) :: value, expected, diff, fac
+    fac = 1.0d0
+    expected = 0.0d0 * fac
+    value = SPEED_OF_LIGHT_IN_VACUUM%to_real(fac, uncertainty=.true.) * fac
+    diff = expected - value
+    call check(error, diff, 0.0d0)
+    if (allocated(error)) return
+end subroutine
+
+subroutine test_U_STANDARD_ACCELERATION_OF_GRAVITY(error)
+    implicit none
+    type(error_type), allocatable, intent(out) :: error 
+    real(dp) :: value, expected, diff, fac
+    fac = 1.0d0
```

Comment:
Suggestion: `fac` could be a `parameter` at the level of the `module`. So it should declared only once.

Reply:
I changed `fac` as a module level variable. I  did not declare it as a parameter because I need to able to change according to the values that are being tested.